tpm2/Makefile

```diff
@@ -100,25 +124,30 @@ pre:
 	rm -f include/*.h~
 	mkdir -p $(BIN_DIR)
 	mkdir -p $(OBJECT_DIRS)
-	mkdir -p $(DOC_DIR)
+	mkdir -p $(DOC_DIRS)
+	mkdir -p $(LOGGER_LIB_DIR)
 
-doc: $(HEADER_FILES) $(UTIL_SOURCES) $(TOOL_SOURCES) $(SEAL_SOURCES) $(UNSEAL_SOURCES) $(GETKEY_SOURCES) Doxyfile
+doc: $(HEADER_FILES) $(UTIL_SOURCES) $(TOOL_SOURCES) $(SEAL_SOURCES) $(UNSEAL_SOURCES) $(GETKEY_SOURCES) $(LOGGER_HEADERS) $(LOGGER_SOURCES) Doxyfile
```

Comment:
Since 'doc' is not a file-based target, it should be phony (.PHONY: doc). If it is not, when the Makefile sees that the doc directory exists, it reports that it is up to date, even though no documentation has been generated or there is a need to generated changed documentation.